examples/stable_diffusion_xl/controlnet.md

```diff
@@ -96,13 +104,122 @@ You can check all arguments description by running `python demo/sampling_without
 <div align="center">
 <img src="https://github.com/HaoyangLee/mindone/assets/20376974/ecdc18c9-36bf-4d49-b7b2-6216400f1d5a" width=30% />
 <img src="https://github.com/HaoyangLee/mindone/assets/20376974/00f0ed2c-c078-4ed3-bee6-cb8c66fb36fd" width=30% />
-<img src="https://github.com/HaoyangLee/mindone/assets/20376974/31da3e94-4633-4506-b2eb-cfd65001d3e8" width=30% />
+<img src="https://github.com/zhtmike/mindone/assets/20376974/28cf7e42-3f4c-4ee8-b4e0-e9a0f8244c43" width=30% />
 </div>
 <p align="center">
 <em> Fig 5. From left to right: raw image - extracted canny edge - inference result. </em>
 </br>
 <em> Prompt: "beautiful bird standing on a trunk, natural color, best quality, extremely detailed". </em>
 </p>
 
+## Training
+
+### Prepare init model weight
+**Step1**: Convert SDXL-base-1.0 model weight from Diffusers to MindONE, refer to [here](../../GETTING_STARTED.md#convert-pretrained-checkpoint). Get `sd_xl_base_1.0_ms.ckpt`.
+
+**Step2**:
+
+```shell
+cd tools/controlnet_conversion
+python init_weight.py
+```
+
+The parameters of `zero_conv`, `input_hint_block` and `middle_block_out` blocks are randomly initialized in ControlNet. Other parameters of ControlNet are copied from SDXL pretrained weight `sd_xl_base_1.0_ms.ckpt` (referring to [here](GETTING_STARTED.md#convert-pretrained-checkpoint)).
+
+### Prepare dataset
+
+We use [Fill50k dataset](https://huggingface.co/datasets/HighCWu/fill50k) to train the model to generate images following the edge control. The directory struture of Fill50k dataset is shown below.
+
+```text
+DATA_PATH
+  ├── prompt.json
+  ├── source
+  │   ├── 0.png
+  │   ├── 1.png
+  │   └── ...
+  └── target
+      ├── 0.png
+      ├── 1.png
+      └── ...
+```
+
+Images in `target/` are raw images. Images in `source/` are the canny edge/segementation/other control images extracted from the corresponding raw images. For example, `source/img0.png` is the canny edge image of `target/img0.png`.
+
+`prompt.json` is the annotation file with the following format.
+
+```json
+{"source": "source/0.png", "target": "target/0.png", "prompt": "pale golden rod circle with old lace background"}
+{"source": "source/1.png", "target": "target/1.png", "prompt": "light coral circle with white background"}
+{"source": "source/2.png", "target": "target/2.png", "prompt": "aqua circle with light pink background"}
+{"source": "source/3.png", "target": "target/3.png", "prompt": "cornflower blue circle with light golden rod yellow background"}
+```
+
+> Note: if you want to use your own dataset for training, please follow the directory and file structure shown above.
+
+### Launch training
+
+Please refer to [`scripts/run_train_base_controlnet.sh`](scripts/run_train_base_controlnet.sh).
+
+```shell
```

Comment:
`nohup` and  `> train.log 2>&1 &` is not nessesary. Users may not use or install nohup

Reply:
Modified